src/ansys/fluent/core/quantity.py

```diff
@@ -1,1369 +1,869 @@
+                )
 
-        Parameters
-        ----------
-        unit_sys: str
-            System name
-        """
-        self._unit_system = unit_sys.upper()
+                # Recursively parse composition unit string
+                si_unit_str, si_multiplier, si_offset = self.si_data(
+                    unit_str=self._derived_units[unit_term]["composition"],
+                    power=unit_term_power,
+                    si_unit_str=si_unit_str,
+                    si_multiplier=si_multiplier,
+                    si_offset=si_offset,
+                )
 
-        if self._unit_system not in UnitSystem._supported_unit_sys:
-            raise ValueError(
-                "Unsupported unit system, only 'SI', 'CGS', 'BT' is allowed."
-            )
+        return self.condense(si_unit_str), si_multiplier, si_offset
 
-    def _get_unit_from_dim(self, dim_list: list) -> str:
-        """
-        Get unit from dimensions.
+    def si_conversion(
+        self,
+        value: float,
+        type: str,
+        si_multiplier: float,
+        si_offset: float,
+        reverse: bool = False,
+    ) -> float:
+        """Performs SI conversion based on quantity type.
 
         Parameters
         ----------
-        dim_list: list
-            List of dimensions
+        value : float
+            Real value of quantity.
+        type : str
+            Quantity type.
+        si_multiplier : float
+            SI factor of quantity object.
+        si_offset : float
+            SI offset of quantity object.
+        reverse : bool
+            SI conversion direction. Setting reverse to `True` performs the inverse of a conversion.
 
         Returns
         -------
-        unit: str
-            Unit using dimensions
+        : float
+            SI value of quantity.
         """
-        unit = ""
-
-        base_units = UnitSystem._dim_to_unit_sys_map[self._unit_system]
-        for key, power in zip(_UnitsTable.dimension_order.values(), dim_list):
-            spacechar = " " if len(unit) > 0 else ""
-            if power == 1:
-                unit += spacechar + base_units[key]
-            elif power != 0:
-                unit += (
-                    spacechar
-                    + base_units[key]
-                    + "^"
-                    + str(int(power) if power.is_integer() else power)
-                )
-        return unit
+        # Perform conversion from SI value
+        if reverse:
+            return (
+                ((value / si_multiplier) - si_offset)
+                if type == "Temperature"
+                else (value / si_multiplier)
+            )
 
-    def convert(self, quantity):
-        """
-        Convert to a system.
+        # Perform conversion to SI value
+        return (
+            ((value + si_offset) * si_multiplier)
+            if type == "Temperature"
+            else (value * si_multiplier)
+        )
+
+    def condense(self, unit_str: str) -> str:
```

Comment:
Used throughout the module: is the `_str` in `unit_str` necessary? Maybe just change it to `units` (as in "these are the __units__ of this number"), makes it easier to use (and it is apparently used a lot in the tests), and we could probably guess it is a string (what else would `units` be in this case?).